### PR TITLE
feat: report modal and modal footer

### PR DESCRIPTION
**Pull request by el**
## Changes
Adds support for Modal footer and replicates the Report Article modal

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-730 #done

### CODE REVIEW

File: packages/shared/src/components/modals/ReportPostModal.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import {
   ReportButtonsCenter,
```
Comment (el) [inline packages/shared/src/components/modals/ReportPostModal.tsx:12]: These will not be necessary anymore

File: packages/shared/src/components/modals/ReportPostModal.tsx [hunk 2]
```diff
@@ -42,6 +43,53 @@ export default function RepostPostModal({
...
+  // return (
+  //   <Modal
+  //     isOpen
+  //     kind={Modal.Kind.FlexibleCenter}
+  //     size={Modal.Size.Small}
+  //     onRequestClose={props.onRequestClose}
+  //     {...props}
+  //   >
+  //     <Modal.Header title="Report article" />
+  //     <Modal.Body>
```
Comment (el) [inline packages/shared/src/components/modals/ReportPostModal.tsx:55]: Although mostly similar, there are some classname differences inside

**Comment by el** [comment]
Previous:
image


Now:
image

Actual design:
https://www.sketch.com/s/2867ec63-0966-4647-b1d7-b6fe696ba251/a/l1y1RKe

File: packages/shared/src/components/modals/ReportPostModal.tsx [hunk 3]
```diff
@@ -42,6 +43,53 @@ export default function RepostPostModal({
...
+  //       />
+  //       <Checkbox
+  //         ref={inputRef}
+  //         name="blockSource"
+  //         className="self-center font-normal"
+  //       >
+  //         Don&apos;t show articles from {post?.source?.name}
+  //       </Checkbox>
+  //     </Modal.Body>
+  //     <Modal.Footer>
```
Comment (rebelchris) [inline packages/shared/src/components/modals/ReportPostModal.tsx:81]: How do we plan to set the multiple buttons?

File: packages/shared/src/components/modals/common/ModalFooter.tsx [hunk 4]
```diff
@@ -5,5 +5,9 @@ export type ModalFooterProps = {
 };
 
 export function ModalFooter({ children }: ModalFooterProps): ReactElement {
-  return <footer>{children}</footer>;
+  return (
+    <footer className="flex gap-3 justify-end items-center py-4 px-2 w-full h-14 border-t border-theme-divider-tertiary">
```
Comment (rebelchris) [inline packages/shared/src/components/modals/common/ModalFooter.tsx:9]: I see the justify-end here, but not sure that's the case with multiple ones right?
Also pretty sure I know the answer, does every footer have a border?

**Review by rebelchris — COMMENTED** [review]
Some questions regarding generic modal options

File: packages/shared/src/components/modals/common/ModalFooter.tsx (hunk 4 above)
Comment (el) [reply packages/shared/src/components/modals/common/ModalFooter.tsx:9]: Footer seems to be right-aligned:
https://www.sketch.com/s/6d2c8a99-2779-4674-87f4-43717dc23eec/a/EL3nQdO

Comment (rebelchris) [reply packages/shared/src/components/modals/common/ModalFooter.tsx:9]: Yes I see, just the one that has it centered (but see you raised this already)
And the markdown footer that's different right?

Comment (el) [reply packages/shared/src/components/modals/common/ModalFooter.tsx:9]: Yeap, I am workin on it as part of the header tabs

**Review by rebelchris — APPROVED** [review]

**Pull request merged**